backend/src/main/java/com/google/rolecall/services/SectionServices.java

```diff
@@ -279,10 +279,40 @@ public Section editSection(SectionInfo newSection) throws EntityNotFoundExceptio
    *    in the database.
    */
   public void deleteSection(int id) throws EntityNotFoundException, InvalidParameterException {
+    checkDeleteOk(id, false);
+    Section section = getSection(id);
+
+    if(section.getType() == Section.Type.SUPER) {
+      // Check that all children are OK to delete
+      for(Position position: section.getPositions()) {
+        Integer childId = position.getSiblingId();
+        if(childId != 0 && childId > 0) {
+          Optional<Section> childQuery = sectionRepo.findById(childId);
+          if(!childQuery.isEmpty()) {
+            checkDeleteOk(childId, true);
+          }
+        }
+      }
+      // If none of the children have a deleate problem, we can delete
+      for(Position position: section.getPositions()) {
+        Integer childId = position.getSiblingId();
+        if(childId != 0 && childId > 0) {
+          Optional<Section> childQuery = sectionRepo.findById(childId);
+          if(!childQuery.isEmpty()) { 
+            sectionRepo.deleteById(childId);
+          }
+        }
+      }
+    }  
+    sectionRepo.deleteById(id);
+  }
+
+  // Utility functions
+  private void checkDeleteOk(int id, boolean superChildIsOk) throws EntityNotFoundException, InvalidParameterException {
```

Comment:
I feel like the `Ok` suffix in the function name and in the `superChildIsOk ` is ambiguous. Reading this function on its own, it could mean:
"has been deleted successfully"
"some object has proper structure"
"operation is ready to be performed"

I suggest changing it to something more meaningful, like `assertDeleteAllowed()` or maybe something else that makes more sense.

Reply:
Done.